app/src/main/kotlin/com/wire/android/navigation/SavedStateViewModel.kt

```diff
@@ -0,0 +1,22 @@
+package com.wire.android.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.navigation.NavBackStackEntry
+
+open class SavedStateViewModel(open val savedStateHandle: SavedStateHandle) : ViewModel()
```

Comment:
Does this class is meant to be inherited from every `ViewModel` that needs to handle a `SavedStateHandle` ?
As a suggestion, maybe we could add a doc/comment, so in the future we know when to use it :D

Reply:
For every `ViewModel` that wants to receive back navigation arguments, so there can be also regular `ViewModel` objects that have `SavedStateHandle`, they just won't get the back navigation arguments, so for consistency and safety it would be better to just use `SavedStateViewModel`. 😌 
I was trying to do it without providing a new class, but I really didn't want to use reflection or any other sorcery. 😄 
I added a commend and updated this [doc about Jetpack Compose & Navigation](https://wearezeta.atlassian.net/wiki/spaces/AR/pages/547458852/Jetpack+Compose+Navigation).